var/spack/repos/builtin/packages/gptune/package.py

```diff
@@ -0,0 +1,173 @@
+    depends_on('cmake@3.3:', type='build')
+    depends_on('jq', type='run')
+    depends_on('blas', type='link')
+    depends_on('lapack', type='link')
+    depends_on('scalapack', type='link')
+    depends_on('py-setuptools', type='build')
+    depends_on('py-ipyparallel', type=('build', 'run'))
+    depends_on('py-numpy', type=('build', 'run'))
+    depends_on('py-pandas', type=('build', 'run'))
+    depends_on('py-joblib', type=('build', 'run'))
+    depends_on('py-scikit-learn', type=('build', 'run'))
+    depends_on('py-matplotlib', type=('build', 'run'))
+    depends_on('py-pyyaml', type=('build', 'run'))
+    depends_on('py-scikit-optimize@master+gptune', type=('build', 'run'))
+    depends_on('py-gpy', type=('build', 'run'))
+    depends_on('py-lhsmdu', type=('build', 'run'))
+    depends_on('py-hpbandster', type=('build', 'run'))
+    depends_on('py-opentuner', type=('build', 'run'))
+    depends_on('py-ytopt-autotune@1.1.0', type=('build', 'run'))
+    depends_on('py-filelock', type=('build', 'run'))
+    depends_on('py-requests', type=('build', 'run'))
+    depends_on('py-cython', type=('build', 'run'))
+    depends_on('py-pyaml', type=('build', 'run'))
+    depends_on('py-mpi4py@3.0.3:', type=('build', 'run'))
+    depends_on('pygmo', type=('build', 'run'))
+    depends_on('openturns', type=('build', 'run'))
+
+    depends_on('superlu-dist@develop', when='+app', type=('build', 'run'))
+
+    conflicts('openmpi@:3')
+
+    def cmake_args(self):
+        spec = self.spec
+        fc_flags = []
+        if '%gcc@10:' in spec or self.spec.satisfies('%apple-clang@11:'):
+            fc_flags.append('-fallow-argument-mismatch')
+
+        args = [
+            '-DTPL_BLAS_LIBRARIES=%s' % spec['blas'].libs.joined(";"),
+            '-DTPL_LAPACK_LIBRARIES=%s' % spec['lapack'].libs.joined(";"),
+            '-DTPL_SCALAPACK_LIBRARIES=%s' % spec['scalapack'].
+            libs.joined(";"),
+            '-DCMAKE_Fortran_FLAGS=' + ''.join(fc_flags),
+            '-DBUILD_SHARED_LIBS=ON',
+        ]
+
+        return args
+
+    examples_src_dir = 'examples'
+    src_dir = 'GPTune'
+    nodes = 1
+    cores = 4
+
+    @run_after('install')
+    def cache_test_sources(self):
+        """Copy the example source files after the package is installed to an
+        install test subdirectory for use during `spack test run`."""
+        self.cache_extra_test_sources([self.examples_src_dir, self.src_dir])
+
+    def test(self):
+        spec = self.spec
+        comp_name = self.compiler.name
+        comp_version = str(self.compiler.version).replace('.', ',')
+        test_dir = join_path(self.install_test_root, self.examples_src_dir)
+
+        if '+app' in spec:
+            superludriver = join_path(spec['superlu-dist'].prefix.bin, 'pddrive_spawn')
+            op = ['-r', superludriver, '.']
+            # copy superlu-dist executables to the correct place
+            with working_dir(join_path(test_dir, 'SuperLU_DIST'), create=False):
```

Comment:
Since all of the commands here use `self.run_test()`, you could use the `work_dir` argument instead of enclosing this in `with working_dir()`.

Reply:
Thanks I will change this in the next PR